kernels/sifive_x280/1/bli_invscalv_sifive_x280_intr/bli_invscalv_sifive_x280_intr_complex.c

```diff
@@ -0,0 +1,83 @@
+/*
+
+   BLIS
+   An object-based framework for developing high-performance BLAS-like
+   libraries.
+
+   Copyright (C) 2024, SiFive, Inc.
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are
+   met:
+    - Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    - Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    - Neither the name(s) of the copyright holder(s) nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+   HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+// clang-format off
+#ifdef INVSCALV
+
+INVSCALV(PRECISION_CHAR, void)
+{
+    (void)cntx;
+    const DATATYPE* restrict alpha = alpha_;
+    DATATYPE* restrict x = x_;
+
+    if (n <= 0) return;
+    if (PASTEMAC(PRECISION_CHAR, eq1)(*alpha)) return;
+    if (PASTEMAC(PRECISION_CHAR, eq0)(*alpha)) return;
+
+    DATATYPE alpha_conj_inv;
+    PASTEMAC(PRECISION_CHAR, copycjs)(conjalpha, *alpha, alpha_conj_inv); 
+    PASTEMAC(PRECISION_CHAR, inverts)(alpha_conj_inv);
+
+    size_t avl = n;
+    while (avl) {
+        size_t vl = VSETVL(PREC, LMUL)(avl);
+        RVV_TYPE_FX(PREC, LMUL, 2) xvec;
+
+        if (incx == 1)
+            xvec = VLSEG2_V_F(PREC, LMUL, 2)((BASE_DT*) x, vl);
+        else
+            xvec = VLSSEG2_V_F(PREC, LMUL, 2)((BASE_DT*) x, 2 * FLT_SIZE * incx, vl);
+
+        RVV_TYPE_F(PREC, LMUL) xvec_real = VGET_V_F(PREC, LMUL, 2)(xvec, 0);
+        RVV_TYPE_F(PREC, LMUL) xvec_imag = VGET_V_F(PREC, LMUL, 2)(xvec, 1);
+        RVV_TYPE_F(PREC, LMUL) yvec_real, yvec_imag;
+
+        VCMUL_VF(PREC, LMUL, yvec_real, yvec_imag, xvec_real, xvec_imag, alpha_conj_inv.real, alpha_conj_inv.imag, vl);
+
+        RVV_TYPE_FX(PREC, LMUL, 2) yvec = VUNDEFINED_FX(PREC, LMUL, 2)();
```

Comment:
You are trying to assemble a complex vector in interleaved format. Based on my understanding, the counterpart to the desired `vssseg2e32.v` instruction are two `VSET_V_F()` intrinsics.

Shouldn't it be enough to just declare the variable?
```c++
RVV_TYPE_FX(PREC, LMUL, 2) yvec;
```

Reply:
If we only declare `yvec`, the compiler emits an uninitialized variable warning because `yvec` is called as an argument of `VSET_V_F` on L. 69. I bumped the toolchain again; the latest version should have `vundefined` intrinsics.

Reply:
> the latest version should have `vundefined` intrinsics

In that case we should also fix the `#pragma` hack [that was used](https://github.com/flame/blis/blob/8820f8f91efd32e38e2995e73323656ef767bbd8/kernels/sifive_x280/1/bli_scal2v_sifive_x280_intr/bli_scal2v_sifive_x280_intr_complex.c#L82) in `scal2v` to ignore the warning. This is a different topic than this PR, so feel free to open a new PR afterwards if you prefer.

Reply:
I'll go ahead and make the change in this PR since it's a minor change if no one objects.

Reply:
No objection here